addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -79,11 +81,19 @@ options.registry.TableOfContent = options.Class.extend({
      */
     _generateNav: function (ev) {
         this.options.wysiwyg && this.options.wysiwyg.odooEditor.unbreakableStepUnactive();
+        const navEl = this.$target[0].querySelector('.s_table_of_content_navbar');
         const headingsEls = this.$target.find(this.targetedElements).toArray()
             .filter(el => !el.closest('.o_snippet_desktop_invisible'));
         const areHeadingsEqual = this.oldHeadingsEls.length === headingsEls.length
             && this.oldHeadingsEls.every((el, i) => el.isEqualNode(headingsEls[i]));
-        if (areHeadingsEqual) {
+        const areVisibilityIdsMatching = headingsEls.every((headingEl) => {
+            const visibilityId = headingEl.closest('section').getAttribute('data-visibility-id');
+            const matchingLinkEl = navEl.querySelector(`a[href="#${headingEl.getAttribute('id')}"]`);
+            const matchingLinkVisibilityId = matchingLinkEl ? matchingLinkEl.getAttribute('data-visibility-id') : null;
```

Comment:
is that possible that there is no `matchingLinkEl`? I am ok to keep it, I am asking just to know

Reply:
The first time we drop the `TableOfContents`, `matchingLinkEl` can indeed be null. This situation has been addressed in the code, as demonstrated in the video. Thanks!

https://github.com/odoo/odoo/assets/125345965/602f25a4-f6f5-461f-bf92-f4950281a278